axis.go

```diff
@@ -407,14 +418,63 @@ func (DefaultTicks) Ticks(min, max float64) (ticks []Tick) {
 		if val >= min && val <= max && !found {
 			ticks = append(ticks, Tick{Value: val})
 		}
-		if math.Nextafter(val, val+minorDelta) == val {
-			break
-		}
 		val += minorDelta
 	}
 	return
 }
 
+func removeDuplicates(elements []float64) []float64 {
+	// Uses a map to record duplicates as we find them.
+	encountered := map[float64]bool{}
+	result := []float64{}
+
+	for v := range elements {
+		if encountered[elements[v]] == false {
+			// Records this element as an encountered element.
+			encountered[elements[v]] = true
+			// Appends it to the result slice.
+			result = append(result, elements[v])
+		} // Otherwise, no elements are added as they are duplicates.
+	}
+	// Returns the new slice.
+	return result
+}
+
+func findRightLevelOfPrecision(elements []float64) []float64 {
+
+	var result []float64
```

Comment:
instead of declaring `result` here and reinitializing it in the outer for-loop, just declare it inside that for-loop (and `return nil` at the end of the function.)

Reply:
Changed that. Thanks!